### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by wintan1418**
TODO-LIST-TESTING
In this project, I was asked  to develop unit tests for an existing project: your To-Do list implementation

### CODE REVIEW

File: test/todo.test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    todoDelete !== todoCheck;
+  });
+});
+
+describe('update', () => {
+  let update = 'I am going to the gym';
+  const updateSave = update.id;
+  it('update task from localStorage', () => {
+    update = 'I am going to the gym and to the cinema';
+    updateSave === update;
```
Comment (Berabjesus) [inline test/todo.test.js]: - [ ] Use jest's `toEqual` function instead of `===`

File: test/todo.test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  });
+});
+
+describe('delete', () => {
+  const todo = 'I am going to the cinema';
+  const todoDelete = Project.getAll().lenght;
+  it('deletes task from localStorage', () => {
+    todo.delete;
+    const todoCheck = Project.getAll().lenght;
+    todoDelete !== todoCheck;
```
Comment (Berabjesus) [inline test/todo.test.js]: - [ ] Use jest's `not.toEqual` function instead of `!==`

File: src/js/utils.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
-  return `ID${digits}`;
-}
+export const parser = Range.prototype.createContextualFragment.bind(
+  document.createRange(),
+);
+
+export function randomID() {
+  const digits = String(Math.random()).split('.')[1].substr(0, 10);
+  return `ID${digits}`;
+}
```
Comment (Berabjesus) [inline src/js/utils.js]: - [ ] You can test this function whether it is returning the required output or not

File: src/js/components.js [hunk 4]
```diff
@@ -1,106 +1,106 @@
...
-        </select>
-      </div>
-    </form>
-  `;
-  return todoForm;
-}
+import { format } from 'date-fns';
+import { parser } from './utils';
+
+export function makeTodoCard(todo, index = null) {
```
Comment (Berabjesus) [inline src/js/components.js]: - [ ] You have to test the functions in component.js if they return the right component with the right class or not.
- for example
```
expect(YOUR_NODE_NAME.nodeName === 'SECTION' && YOUR_NODE_NAME.getAttribute('class')).toBe('col-12 col-md-5 d-flex text-dark h-100vh second-section');
```

**Review by Berabjesus — CHANGES_REQUESTED** [review]
# Status: CHANGES REQUIRED :warning: :construction: 
Hey @wintan1418 great start to this milestone. You have made nice tests for some functions and all tests are passing. 👏 

### However, you do have a few changes to carry out before we proceed so please take a look at the things you have to work on and their respective review comments below.

- [ ] Please include instruction in your readme on how to run tests.
- [ ] Create separate test file for each class or module. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/test_each_module_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/70449618/)) and make sure all functions are tested, currently, you have only tested the `todo` and `project` files. But functions in the `components` and even in the `utils` are testable. 
- [ ] Use jest's `toEqual` function instead of `===`
- [ ] remove `.vscode` folder or add it to .gitignore

## **If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help** 
**HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

File: src/js/utils.js (hunk 3 above)
Comment (mahmoud717) [reply src/js/utils.js]: - [ ] This is to return a random number, therefore it's untestable.

**Review by mahmoud717 — APPROVED** [review]
### STATUS: APPROVED ✔️ ✔️ 

**hi @wintan1418**

**Great job with making all of the requirements. the project is approved, you can go ahead and merge this pull request**

**Pull request merged**